fe/fe-core/src/main/java/com/starrocks/sql/optimizer/OptimizerContext.java

```diff
@@ -141,4 +149,17 @@ public List<MaterializationContext> getCandidateMvs() {
     public void addCandidateMvs(MaterializationContext candidateMv) {
         this.candidateMvs.add(candidateMv);
     }
+
+    public void updateUsedMv(List<Table> tables, MaterializedView mv) {
+        List<Table> sortedTables = tables.stream()
+                .sorted((left, right) -> left.getName().compareTo(right.getName())).collect(Collectors.toList());
+        List<MaterializedView> mvs = usedMvs.computeIfAbsent(sortedTables, key -> Lists.newArrayList());
+        mvs.add(mv);
+    }
+
+    public List<MaterializedView> getUsedMvs(List<Table> tables) {
```

Comment:
It looks like not like usedMvs, but like 'searchedMvs'

Reply:
move it into MaterializationContext, and change it to matchedTableLists